pkg/cmd/pr/merge/merge.go

```diff
@@ -167,17 +168,45 @@ func mergeRun(opts *MergeOptions) error {
 			if err != nil {
 				return err
 			}
-			confirm, err := confirmSurvey()
+
+			allowEditMsg := (mergeMethod != api.PullRequestMergeMethodRebase) && !opts.BodyProvided
```

Comment:
We can do that. I explicitly disallowed since it seems like if the user provided the `--body` flag they would have already typed in what they want the body message to me and giving them the option to further edit it would just add confusion. I don't have strong feelings either way though.

Reply:
I'm on the fence about it as well and I can see your reasoning. But, it seems to me that we gain nothing by explicitly hiding the interactive edit option from the user— we just make the interactive mode inconsistent based on input flags, but the user doesn't get a better experience (in my view). So, for the sake for consistency and simplicity, we can allow unconditional edit in interactive mode.